### PR TITLE
fix: [PL-22646]: Invalid branch fixes

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
##### Summary:

Changes are : 
- Invalid branch fixes : `GitContextForm` is always initialised with a given <Repo, branch> or <1st Repo, default branch>.Bug was to push branch used for initialising for all the repo.

- CSS fixes : Loading Icon was in far right.And branch Select dropdown size was changing while loading.

- Making Branch list size as 100 as decided for consistency
- Including `GitContextForm` of common under GitSync coverage

##### Jira Links:
https://harness.atlassian.net/browse/PL-22646
##### Screenshots:

**Before fix as in QA now**
https://user-images.githubusercontent.com/63278928/159774873-cb780830-af74-4609-af75-f88547ceecb0.mov

**After fix:** 

https://user-images.githubusercontent.com/63278928/159774393-289e1d7b-ecf8-44a4-b8ea-16d821ac3e6c.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by sunnykesh-harness** [comment]
retrigger titlecheck

**Comment by sunnykesh-harness** [comment]
retrigger titlecheck

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/10-common/components/GitContextForm/GitContextForm.tsx [hunk 1]
```diff
@@ -196,7 +197,7 @@ const GitContextForm: React.FC<GitContextFormProps<Record<string, any> & GitCont
           }}
           className={css.selectBranchInput}
         />
-        {loadingBranchList && <Icon margin={{ top: 'xsmall' }} name="spinner" />}
+        <Container width={16}>{loadingBranchList && <Icon margin={{ top: 'xsmall' }} name="spinner" />}</Container>
```
Comment (avinashmadhwani02) [inline src/modules/10-common/components/GitContextForm/GitContextForm.tsx:200]: Is Container required? we can give size to icon directly

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/10-common/components/GitContextForm/GitContextForm.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/10-common/components/GitContextForm/GitContextForm.tsx:200]: It was like that before. Giving Icon directly increasing the Branch Select width as seen in screen recording.
It is required to keep the Select width consistent as 54% of modal.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**